File: backend/src/AnimeShop/API/Maintenance.hs
{-# LANGUAGE DataKinds #-}
{-# LANGUAGE TypeOperators #-}
module AnimeShop.API.Maintenance (MaintenanceAPI, maintenanceServer) where

import AnimeShop.Prelude
import AnimeShop.API.Maintenance.Equipment
import AnimeShop.API.Maintenance.EquipmentInstances
import AnimeShop.API.Maintenance.EquipmentInspectionSchedules
import AnimeShop.API.Maintenance.EquipmentInspectionResults

type MaintenanceAPI = "maintenance" :>
  (EquipmentAPI :<|>
   EquipmentInstancesAPI :<|>
   EquipmentInspectionSchedulesAPI :<|>
   EquipmentInspectionResultsAPI)

maintenanceServer :: ServerT MaintenanceAPI App
maintenanceServer =
  equipmentServer :<|>
  equipmentInstancesServer :<|>
  equipmentInspectionSchedulesServer :<|>
  equipmentInspectionResultsServer

File: frontend/src/components/Pages/Maintenance.js
import React from "react";

import Equipment from "../Maintenance/Equipment";
import EquipmentInstances from "../Maintenance/EquipmentInstances";
import EquipmentInspectionSchedules from "../Maintenance/EquipmentInspectionSchedules";
import EquipmentInspectionResults from "../Maintenance/EquipmentInspectionResults";

export function Maintenance() {
    return (
      <div>
        <Equipment/>
        <EquipmentInstances/>
        <EquipmentInspectionSchedules/>
        <EquipmentInspectionResults/>
      </div>
    );
  }
